### PR TITLE
Fix writing global key text after analyzing to library and file metadata

### LINKED ISSUE #13646: Analyzed keys are not saved as text
### Bug Description

Neither in the database nor as metadata.

`Keys::getGlobalKeyText()` in `TrackRecord::setKeys()` returns an empty String.

The database contains an empty string in `library.key` and the file tag is missing.

### Version

main

### OS

_No response_

### PR DESCRIPTION
**Pull request by daschuer**
This fixes #13646 a regression since #11096

Thank you @uklotzde for finding this bug just in time before the 2.5-beta period ends.

 I have also added a unittest.

### CODE REVIEW

File: src/track/keyfactory.cpp [hunk 1]
```diff
@@ -106,7 +106,14 @@ Keys KeyFactory::makePreferredKeys(
...
             pChange->set_frame_position(static_cast<int>(frame));
         }
-        key_map.set_global_key(KeyUtils::calculateGlobalKey(key_changes, totalFrames, sampleRate));
+
+        mixxx::track::io::key::ChromaticKey global_key =
+                KeyUtils::calculateGlobalKey(
+                        key_changes, totalFrames, sampleRate);
+        key_map.set_global_key(global_key);
+        QString global_key_text = KeyUtils::keyToString(
+                global_key, KeyUtils::KeyNotation::ID3v2);
```
Comment (Swiftb0y) [inline src/track/keyfactory.cpp:115]: what implications does hardcoding the notation have? Does this overwrite the user-notation?

Comment (uklotzde) [reply src/track/keyfactory.cpp:115]: Not affected. This text is generated from the underlying data dynamically according to the preferences.

**Comment by uklotzde** [comment]
Thank you. Works as expected.

Affected users need to re-analyze the key to fix the inconsistencies in the database and file tags.

File: src/track/keyfactory.cpp (hunk 1 above)
Comment (daschuer) [reply src/track/keyfactory.cpp:115]: This ensures that we write the ID3v2 to the tags. https://github.com/mixxxdj/mixxx/issues/10890

**Review by acolombier — APPROVED** [review]

File: src/track/keyfactory.cpp (hunk 1 above)
Comment (Swiftb0y) [reply src/track/keyfactory.cpp:115]: Great. thanks for the clarification.

**Review by Swiftb0y — APPROVED** [review]
Thank you. LGTM.

**Comment by fwcd** [comment]
Stumbled upon this after one of my scripts that parses the keys from the db broke. This fixes it for me too! Can the PR be merged?

**Comment by Swiftb0y** [comment]
yup. sorry for forgetting about it.

**Comment by fwcd** [comment]
Could someone take a look at merging 2.5? Working around this bug on main is becoming a bit fiddly...

**Comment by FrankwaP** [comment]
Sorry but I still get empty "key" fields in my database with a compiled Mixxx 2.6.0-alpha (using libkeyfinder).

**Comment by Swiftb0y** [comment]
merging 2.4 to 2.5 there are merge conflicts in wtracktableview I don't understand. @ronso0 could you look into those?

**Comment by ronso0** [comment]
Yup, merging 2.4 now

**Comment by ronso0** [comment]
Uff, this is hard.. I'll open a PR, just in case

**Pull request merged**